pkg/pmem-csi-driver/pmem-csi-driver.go

```diff
@@ -258,6 +278,62 @@ func (pmemd *pmemDriver) registerNodeController() error {
 	return nil
 }
 
+// startScheduler starts the scheduler extender if it is enabled. It
+// logs errors and cancels the context when it runs into a problem,
+// either during the startup phase (blocking) or later at runtime (in
+// a go routine).
+func (pmemd *pmemDriver) startScheduler(ctx context.Context, cancel func(), rs *registryserver.RegistryServer) error {
+	if pmemd.cfg.schedulerListen == "" {
+		return nil
+	}
+
+	resyncPeriod := 1 * time.Hour
+	factory := informers.NewSharedInformerFactory(pmemd.cfg.client, resyncPeriod)
+	pvcLister := factory.Core().V1().PersistentVolumeClaims().Lister()
+	scLister := factory.Storage().V1().StorageClasses().Lister()
+	sched, err := scheduler.NewScheduler(
+		pmemd.cfg.DriverName,
+		scheduler.CapacityViaRegistry(rs),
+		pmemd.cfg.client,
+		pvcLister,
+		scLister,
+	)
+	if err != nil {
+		return fmt.Errorf("create scheduler: %v", err)
+	}
+
+	server := http.Server{
+		Addr:    pmemd.cfg.schedulerListen,
+		Handler: sched,
+	}
+	factory.Start(ctx.Done())
+	cacheSyncResult := factory.WaitForCacheSync(ctx.Done())
+	klog.V(5).Infof("synchronized caches: %+v", cacheSyncResult)
+	for t, v := range cacheSyncResult {
+		if !v {
+			cancel()
+			return fmt.Errorf("failed to sync informer for type %v", t)
+		}
+	}
+
+	go func() {
+		err := server.ListenAndServeTLS(pmemd.cfg.CertFile, pmemd.cfg.KeyFile)
```

Comment:
The elephant in the room here is TLS again: this will run an HTTP server which allows more ciphers than we want...

I've added the new TLS check (PR #545) in this PR. That check should trigger.

Reply:
I force-pushed a version which uses the same `LoadServerTLS` config as for grpc servers.